modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -200,35 +200,17 @@ class RemoveSliceAllOptionalInputsSubgraph : public Subgraph {
 
         slice_id = addNodeToMatch("Slice", inputs);
 
-        setFusedNode("Slice", std::vector<int>{input, starts, ends});
+        setFusedNode("Slice", inputs);
     }
 
-    virtual bool match(const Ptr<ImportGraphWrapper>& net, int nodeId,
-                       std::vector<int>& matchedNodesIds) CV_OVERRIDE {
-        if (Subgraph::match(net, nodeId, matchedNodesIds)) {
-            if (num_inputs_ >= 4) { // with axes
-                // Check if axes are -1 or last axis
-                auto onnx_net = net.dynamicCast<ONNXGraphWrapper>();
-                int shape_size = onnx_net->getTensorShapeSize(matchedNodesIds[slice_id], 0);
-
-                auto axes = extractConstant(net, matchedNodesIds[slice_id], 3);
-                for (size_t i = 0; i < axes.total(); i++) {
-                    const int axis = *(axes.ptr<const int>() + i);
-                    if (axis != -1 && axis != shape_size - 1) {
-                        return false;
-                    }
-                }
-            }
-            if (num_inputs_ == 5) {
-                // Check if steps are 1
-                auto steps = extractConstant(net, matchedNodesIds[slice_id], 4);
-                if (countNonZero(steps != 1)) {
-                    return false;
-                }
-            }
-            return true;
+    virtual void finalize(const Ptr<ImportGraphWrapper>&,
+                          const Ptr<ImportNodeWrapper>& fusedNode,
+                          std::vector<Ptr<ImportNodeWrapper> >&) CV_OVERRIDE
+    {
+        opencv_onnx::NodeProto* node = fusedNode.dynamicCast<ONNXNodeWrapper>()->node;
+        for (int i = num_inputs_; i < 5; ++i) {
+            node->add_input("");
```

Comment:
Doesn't it lose connections to the existing constants (starts, ends, axes and steps)?

Reply:
No, It should append missed connections to `axes` and `steps` pointing to a fake initializer added by https://github.com/opencv/opencv/blob/850ebec135bb7904de4a1b9abbbf32b9937c1c06/modules/dnn/src/onnx/onnx_graph_simplifier.cpp#L71

Reply:
How does this appending happen? What I see here is input names are missed already.

Reply:
For example, `node` which found with 3 real inputs (data, starts, ends) appends by up to 5 inputs with `add_input`:
```
inputs: "data_name", "starts_name", "ends_name", "", ""
```

`net.add_initializer();` adds an empty tensor with an empty name. So the final version of the Slice refers to it's tensor.

I have observed such behavior with `resize_nearest_unfused_opset11_torch1.4` model:
![image](https://github.com/opencv/opencv/assets/25801568/dbef35d3-c275-4fd4-a6ae-a0b4748a887c)

Reply:
Yes, I got you point here. But actually my problem is why this

```.diff
- setFusedNode("Slice", std::vector<int>{input, starts, ends});
+ setFusedNode("Slice", inputs);
```

does not drop the `starts` and `ends` constants.

Reply:
They were added before. Just there is no overload with 5 inputs and vector is used:
```cpp
std::vector<int> inputs{input, starts, ends};
for (size_t i = 3; i < num_inputs_; i++) { // axes and steps
    inputs.push_back(addNodeToMatch(""));
}
```

Reply:
Ahhh, sorry, I missed that. I thought it was `input`.

Reply:
@dkurt is there a way for me to check these fixes locally before merging?

Reply:
@Abdurrahheem, do the following:
```
git remote add dkurt https://github.com/dkurt/opencv
git fetch dkurt adjust_slice_optional_inputs
git checkout dkurt/adjust_slice_optional_inputs
```

Reply:
> @dkurt is there a way for me to check these fixes locally before merging? Wanna make sure if yolox test do pass.

git cherry-pick should also work right? The simplest way is do the same changes as this PR manually, so you don’t need to mess around with git operations.

Reply:
@dkurt just checked test with the new patch. Everything passes related to yolo detectors. Thank you for a quick fix!